### PR TITLE
SEARCH-2149: add customer group ID to support B2B

### PR DESCRIPTION
**Pull request by RuslanKostiv1**
## Description
Pass customer group id (in the scope of `context`) to snowplow

## Related Issue
https://jira.corp.magento.com/browse/SEARCH-2149

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [ ] I have read the **CONTRIBUTING** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.

### CODE REVIEW

File: src/types/schemas/searchInput.ts [hunk 1]
```diff
@@ -11,6 +11,11 @@ export type SearchInputUnit = {
     currentPage: number;
     filter: Array<SearchFilter>;
     sort: Array<SearchSort>;
+    context: QueryContextInput;
+};
+
+export type QueryContextInput = {
+    customerGroup?: string;
```
Comment (aniham) [inline src/types/schemas/searchInput.ts:18]: @RuslanKostiv1 shouldn't we make `context` optional and `customerGroup` required to match the search schema:

https://github.com/adobe/magento-storefront-event-collector/pull/143/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519

Comment (RuslanKostiv1) [reply src/types/schemas/searchInput.ts]: good point, fixed

**Review by aniham — APPROVED** [review]

File: tests/mocks.ts [hunk 2]
```diff
@@ -261,6 +261,7 @@ export const generateSearchInputContext = (
             currentPage: 1,
             filter: [{ attribute: "size", eq: "small" }],
             sort: [{ attribute: "relevance", direction: "DESC" }],
+            context: { customerGroup: "Not logged in" },
```
Comment (priyasbhat) [inline tests/mocks.ts:264]: Can we use `NLI` here, just for consistency?

**Review by priyasbhat — APPROVED** [review]

File: tests/mocks.ts (hunk 2 above)
Comment (RuslanKostiv1) [reply tests/mocks.ts]: updated

**Pull request merged**